tests/test_cmdline.py

```diff
@@ -347,6 +347,76 @@ def test_combine(self):
                        os.path.join(example_json_dir, "different_sorts.json")],
                       expected_output_file="result.json")
 
+    def test_set_sync_marker(self):
+        test_script = textwrap.dedent("""
+            from viztracer import get_tracer
+
+            get_tracer().set_sync_marker()
+            get_tracer().set_sync_marker()
+
+        """)
+
+        def expect_sync_marker(data):
+            self.assertGreater(data['viztracer_metadata'].get('sync_marker'), 0)
+
+        self.template(
+            [sys.executable, "-m", "viztracer", "--ignore_frozen", "-o", "result.json", "cmdline_test.py"],
+            expected_output_file="result.json",
+            expected_stderr="Synchronization marker already set",
+            script=test_script,
+            check_func=expect_sync_marker,
+        )
+
+    def test_align_combine_sync_marker(self):
+        test_script = textwrap.dedent("""
+            import random
+            import time
+            from viztracer import get_tracer
+
+            def func():
+                return 2 * 4
+
+            # will should sleep from 50ms to 100ms
+            time.sleep(random.uniform(0.05, 0.1))
+
+            get_tracer().set_sync_marker()
+            func()
+        """)
+
+        def expect_aligned_to_sync_marker(data):
+
+            funcs = [event for event in data['traceEvents'] if 'ts' in event and event['name'].startswith('func ')]
```

Comment:
Is the reason we use `'func '` here (instead of `func`) that there are other events? Or just to make sure it's `func` function instead of a function starting with func? Random spaces in code sometimes will throw people off. I think the events should be pretty controllable right? Especially you ignored frozen. Can we remove the space or rename the function to something more unique?